client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings.js

```diff
@@ -0,0 +1,75 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+
+/**
+ * Internal dependencies
+ */
+import {
+	getCarrierRegistrationFields
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+import * as api from 'woocommerce/woocommerce-services/api';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import DynamicCarrierAccountSettingsForm from './dynamic-settings-form';
+
+export const DynamicCarrierAccountSettings = ( props ) => {
+	const [carrierRegistrationFields, setCarrierRegistrationFields] = useState([]);
+
+	useEffect(() => {
+		const fetchRegistrationFields = async () => {
+			const registrationFields = await api.get(props.siteId, api.url.shippingCarrierTypes());
+			setCarrierRegistrationFields(registrationFields.carriers);
+		}
+		fetchRegistrationFields();
+	}, [props.siteId]);
+
+	if (!carrierRegistrationFields || carrierRegistrationFields.length < 1) {
+		return (
+			<div>Loading...</div>
+		);
+	}
+
+	const supportedCarriers = carrierRegistrationFields.filter(carrier => carrier.type === props.carrier);
+
+	if (!supportedCarriers || supportedCarriers.length === 0) {
```

Comment:
In which case `supportedCarriers` would be falsy? `Array.filter` always returns an array.
```suggestion
	if (!supportedCarriers.length === 0) {
```